server/channels/app/app_iface.go

```diff
@@ -102,11 +117,36 @@ type AppIface interface {
 	// CreateGuest creates a guest and sets several fields of the returned User struct to
 	// their zero values.
 	CreateGuest(c request.CTX, user *model.User) (*model.User, *model.AppError)
+	// CreateOAuthUser creates a user from OAuth information. The user is created with roles determined by the
+	// team and channel membership defaults set by the system admin.
+	CreateOAuthUser(c request.CTX, service string, userData io.Reader, teamID string, tokenUser *model.User) (*model.User, *model.AppError)
+	// CreatePasswordRecoveryToken creates a token that can be used to reset a user's password.
+	// The token is valid for PasswordRecoverExpiryTime duration.
+	// Returns the token if successful, or an error if the token could not be created.
+	CreatePasswordRecoveryToken(rctx request.CTX, userID, email string) (*model.Token, *model.AppError)
 	// CreateUser creates a user and sets several fields of the returned User struct to
 	// their zero values.
 	CreateUser(c request.CTX, user *model.User) (*model.User, *model.AppError)
+	// CreateUserAsAdmin creates a new user with system admin privileges.
+	// Returns the created user and nil on success, or nil and an error on failure.
```

Comment:
Curious to understand why nil is returned here

Reply:
That is an standard pattern in the go code, you return (value, error) if success, you return (value, nil), if error, you return (nil, error).

Reply:
Thanks, @jespino!